test/dlc_test.json

```diff
@@ -0,0 +1,329 @@
+[ {
+  "testInputs" : {
+    "params" : {
+      "oracleInfo" : {
+        "publicKey" : "a78bf5ed2841a93d4dcc1b719ebbcc90c7648746e5bd55fe7806494a93167c39",
+        "nonce" : "d2d6cb7049ef4edc2d98aa3e235e8d542573cf0c9a735516b9936aba52ae693e"
+      },
+      "contractInfo" : {
+        "8873e27fd5804a500792d3b197ec83e3d13311e161c431b223657b64ec78e355" : 0,
+        "9fc2710e301f8700a8a4c6933366c52045f23a59c92eaf17b0c3d6413861d068" : 200000000
+      },
+      "contractMaturityBound" : 100,
+      "contractTimeout" : 200,
+      "feeRate" : 5
+    },
+    "offerParams" : {
+      "collateral" : 100000000,
+      "fundingInputTxs" : [ {
+        "tx" : "02000000000100c2eb0b000000001600145845070bc9e99b0db44039e0567e32408a1ff63600000000",
+        "idx" : 0,
+        "inputKey" : "5c81759faa564e987ce7b658abb0afd593d06ce28d5824d6d630d8295910c707"
+      } ],
+      "changeAddress" : "bcrt1qu7d4c4vhpkf8vpgadfcljw724jcff03zjt5rm2",
+      "fundingPrivKey" : "61f1a2e40d7e50090fc879e4cbb816467c764b303ac9bb5ddccefbb1685b5b5e",
+      "payoutAddress" : "bcrt1qxc6egxhnntdr8h53lhrzqfpxq02hh2yl82newd"
+    },
+    "acceptParams" : {
+      "collateral" : 100000000,
+      "fundingInputTxs" : [ {
+        "tx" : "02000000000100c2eb0b000000001600148a04916238f4c4661637807a37c30d8e17e77f7a00000000",
+        "idx" : 0,
+        "inputKey" : "e2d7b500267e5d1656f2f2fc0fd0580a48f525c0aa537d8b356c1793e9626131"
+      } ],
+      "changeAddress" : "bcrt1q4kw7nkyq42u6nnnhey3ylagpjscq7kwr4f2ufp",
+      "fundingPrivKey" : "5391f9340bdd7c194023d31176b6aab334593a9afe5d3356987d74b2833f2e59",
+      "payoutAddress" : "bcrt1qx9f936etaa43qxnu0ss7zppca296xglzt9z7wv"
+    }
+  },
+  "offer" : "fda71afd015d000f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e2206fda710508873e27fd5804a500792d3b197ec83e3d13311e161c431b223657b64ec78e35500000000000000009fc2710e301f8700a8a4c6933366c52045f23a59c92eaf17b0c3d6413861d068000000000bebc200fda71240a78bf5ed2841a93d4dcc1b719ebbcc90c7648746e5bd55fe7806494a93167c39d2d6cb7049ef4edc2d98aa3e235e8d542573cf0c9a735516b9936aba52ae693e03f12e6c13142cc488b4f3d31b45be115d9637f6ccf4b4e3f1223a1a71052889d11600143635941af39ada33de91fdc620242603d57ba89f0000000005f5e1000001fda71437002902000000000100c2eb0b000000001600145845070bc9e99b0db44039e0567e32408a1ff6360000000000000000ffffffff006c0000160014e79b5c55970d9276051d6a71f93bcaacb094be22000000000000000000000064000000c8",
```

Comment:
I can't figure out why your message starts with `fd`. I think it's supposed to start with the message type which is the following `a71a`?

Reply:
Check the definition of BigSizeUInt here: https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#appendix-a-bigsize-test-vectors